README.md

```diff
@@ -25,32 +15,84 @@ Install as usual, see
 CONFIGURATION
 -------------
 
-The module has a simple configuration screen, where you provide Server Key and
-Firebase's endpoint.
+The module has a simple configuration form, where you provide Server Key and
+Sender ID.
+
+HOW TO USE
+-----------
 
-* In order to send the push notification:
+In order to send the push notification message with data:
 
 ```php
 // Token is generated by app. You'll have to send the token to Drupal.
 $fakeToken = 'e3vUiwcvkpY:APA91bEDZzKTIkaL0e-UTwiV6EGi1m5J5PrDMxejm6-d85vdwAgd';
-\Drupal::service('firebase.notification')->send($fakeToken, [
+$messageService = \Drupal::service('firebase.message');
+$messageService->addRecipients($fakeToken);
+$messageService->addNotification([
   'title' => 'Title goes here',
   'body' => 'Body goes here',
-  'data' => [
-    'score' => '3x1',
-    'date' => '2017-10-10',
-    'optional' => 'Data is used to send silent pushes. Otherwise, optional.',
-  ],
+  'badge' => 1,
   'icon' => 'optional-icon',
   'sound' => 'optional-sound',
   'click_action' => 'optional-action',
-  'badge' => 1,
 ]);
+$messageSerivce->addData([
+  'score' => '3x1',
+  'date' => '2017-10-10',
+  'optional' => 'Data is used to send silent pushes. Otherwise, optional.',
+]);
+$messageService->setOptions(['priority' => 'normal']);
```

Comment:
no way to set single option?

Reply:
this means consumer's code will need to get options, change/add single property and set options

Reply:
As I explained to you on call, you can add/change single option, because body of request it's property of FirebaseServiceBase